core/orchestrator.go

```diff
@@ -478,25 +490,30 @@ func (rt *RemoteTranscoder) Transcode(fname string, profiles []ffmpeg.VideoProfi
 	case <-ctx.Done():
 		// XXX remove transcoder from streams
 		rt.eof <- struct{}{}
-		return [][]byte{}, fmt.Errorf("Remote transcoder took too long")
+		return [][]byte{}, fmt.Errorf("Remote transcoder %s took too long", rt.addr)
 	case chanData := <-taskChan:
-		glog.Info("Successfully received results from remote transcoder ", len(chanData.Segments))
+		glog.Infof("Successfully received results from remote transcoder=%s segments=%d taskId=%d fname=%s",
+			rt.addr, len(chanData.Segments), taskId, fname)
 		return chanData.Segments, chanData.Err
 	}
-	return [][]byte{}, fmt.Errorf("Unknown error")
 }
 
 func NewRemoteTranscoder(n *LivepeerNode, stream net.Transcoder_RegisterTranscoderServer, capacity int) *RemoteTranscoder {
-	return &RemoteTranscoder{
+	t := &RemoteTranscoder{
 		node:     n,
 		stream:   stream,
 		eof:      make(chan struct{}, 1),
 		capacity: capacity,
+		addr:     "unknown",
+	}
+	if p, ok := peer.FromContext(stream.Context()); ok {
```

Comment:
Same question as earlier; under what circumstances is `peer.FromContext` not-`ok`?